templates/create-daml-app/daml/Setup.daml

```diff
@@ -1,10 +1,28 @@
 module Setup where
 
-import DA.Foldable
+import DA.Action (unless)
+import DA.Foldable (forA_)
+import qualified DA.Text as T
 import Daml.Script
 
 setup : Script ()
 setup = do
-  forA_ [("Alice", "alice"), ("Bob", "bob"), ("Charlie", "charlie")] $ \(displayName, userName) -> do
-    p <- allocatePartyWithHint displayName (PartyIdHint ("p" <> userName))
-    createUser (User userName (Some p)) [CanActAs p]
+  let displayNames = ["Alice", "Bob", "Charlie"]
+  let userIds = map toUserId displayNames
+  users <- listUsers
+  -- Only allocate users if they do not already exist.
+  -- This is required for hot reload to work since it
+  -- reruns the start script.
+  unless (all (\t -> t `elem` map (.id) users) userIds) $
```

Comment:
This would mean we'll try to reallocate all users if only one is missing. This would cause an error for the users that have already been allocated. I'd prefer having the condition to allocate to be per user via `getUser`.

Reply:
good point, I changed it as you suggest